### ISSUE TITLE
Maybe a Bug: XYZToENU/EllToENU Always Uses the First Image as Reference Frame

### ISSUE THREAD

**Issue by StonerLing** [colmap/colmap#3055]
Since a comment was left, I am unclear about the consideration for using the input parameters lat0 and lon0 to perform the rotation transformation from ECEF to ENU, while fixing the first image as the origin point.

https://github.com/colmap/colmap/blob/b7fc9ffd278592c21fa28ba16e4efc4ee3359108/src/colmap/geometry/gps.cc#L211-L214

Should it use:
```cpp
  Eigen::Vector3d ref_xyz = EllToXYZ({Eigen::Vector3d(lat0, lon0, alt0)}).front();
  // Convert ECEF to ENU coords. (w.r.t. ECEF ref == xyz[0])
  for (size_t i = 0; i < xyz.size(); ++i) {
    enu[i] = R * (xyz[i] - ref_xyz);
  }
```

And we add a additional parameter alt0 similar to `ENUToEll`/`ENUToEll`.

Otherwise, single input `EllToENU({lat, lon, alt}, lat0, lon0)` will always return `(0,0,0)`.

**Comment by ahojnnes** [comment]
I have never used this functionality myself and it was admittedly back then also not reviewed as thoroughly as I would want to today. There seems to be another bug in that the rotation is computed using the geocentric instead of the geodetic latitude?

**Comment by StonerLing** [comment]
Yes, the rotation must be computed using the geodetic latitude rather than the geocentric latitude. But this bug may occur when calling this function, which entails passing inaccurate lat0 and lon0 values.

**Comment by ahojnnes** [comment]
Right, I guess, this would benefit from some better documentation.

**Comment by xdw-123** [comment]
@StonerLing I really feel the author's suggestion and yours. Our usage scenarios are the same. In your branch, you only did XYZToENU/EllToENU. Do you still use GPS information from drone images to correct the bending problem during the reconstruction process and the alignment problem of the sparse reconstructed model? We look forward to your reply.

**Comment by ahojnnes** [comment]
This is now fixed.
